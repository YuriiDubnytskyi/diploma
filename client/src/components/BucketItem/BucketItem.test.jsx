import React from "react";
import BucketItem from "./BucketItem";

const setUp = (props) => shallow(<BucketItem {...props} />);

describe("Test HTML in BucketItem", () => {
    it("Should contain BucketItem with props", () => {
        const component = setUp({ name: "Nokia", price: "777", count: "3" });
        expect(component.find(".item-bucket__title").text()).toEqual("Назва товару - Nokia");
        expect(component.find(".item-bucket__price").text()).toEqual("Ціна - 777");
        expect(component.find(".btn-bucket").at(1).text()).toEqual("3");
        expect(component).toMatchSnapshot();
    });
});
