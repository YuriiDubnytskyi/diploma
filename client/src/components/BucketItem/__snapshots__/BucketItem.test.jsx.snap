// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test HTML in BucketItem Should contain BucketItem with props 1`] = `
<div
  className="bucket-list__item item-bucket"
>
  <img />
  <div
    className="item-bucket__info"
  >
    <h3
      className="item-bucket__title"
    >
      Назва товару - 
      Nokia
    </h3>
    <p
      className="item-bucket__price"
    >
      Ціна - 
      777
    </p>
  </div>
  <div
    className="item-bucket__btn"
  >
    <button
      className="btn-bucket"
    >
      +
    </button>
    <button
      className="btn-bucket"
    >
      3
    </button>
    <button
      className="btn-bucket"
    >
      -
    </button>
    <button
      className="btn-bucket"
    >
      X
    </button>
  </div>
</div>
`;
